solr/modules/hdfs/src/test/org/apache/solr/hdfs/cloud/SharedFileSystemAutoReplicaFailoverTest.java

```diff
@@ -31,6 +31,7 @@
 import java.util.concurrent.Future;
 import java.util.concurrent.SynchronousQueue;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
```

Comment:
I haven't been able to make this test pass, same with and without the fix. It is flagged `@BadApple` for some time.
My understanding is it was testing the "autoAddReplica" property, which is not supported anymore since removal of autoscaling framework.
See https://github.com/apache/solr/commit/cc0c111949d5039a0c7cb67cad55c63e2f761298#diff-ec4c71bca5a91aa5a49e509688891dc2d21fdda62981add8fb240b256e8d1a80L165
Should this test class just be deleted?

Reply:
autoAddReplica was ported to work without the autoscaling framework; it survived the deletion.  I recommend not removing the test right now; deserves a separate discussion.